packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -263,7 +276,17 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
             assert(batch[batch.length - 1].metadata?.batch === false, 0x3cb /* no end batch markup */);
         }
 
-        this.connectionManager.sendMessages(batch);
+        // TODO: revisit
+        if (batch.length >= 2) {
+            this.connectionManager.sendMessages([{
+                type: GroupedBatchOpType,
+                clientSequenceNumber: batch[0].clientSequenceNumber, // TODO: doesn't matter what we put here because we forget this message
```

Comment:
your comment is not entirely correct, as protocol cares about clientSequenceNumbers - there should be no gaps. For that reason, we either can't assign clientSequenceNumber to each message in the batch (as that will create gaps when observing ops on the wire), OR we have to virtualize them

Reply:
I thought the client sequence number gaps are checked by client? Does the service look at it and do some validation, or is it just treated as some metadata? Or does it matter from the perspective of the connection manager (since it looks to be keeping track)?

Reply:
Currently there are two purposes for clientSequenceNumber concept:
- service validates that all messages for given connection have consecutive numbers
- client uses them to identify messages (i.e. as identity mechanism)